src/frontend/planner_test/tests/testdata/temporal_join.yaml

```diff
@@ -0,0 +1,80 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- name: Left join type for temporal join
+  sql: |
+    create table stream(id1 int, a1 int, b1 int) APPEND ONLY;
+    create table version(id2 int, a2 int, b2 int, primary key (id2));
+    select id1, a1, id2, a2 from stream left join version FOR SYSTEM_TIME AS OF NOW() on id1= id2
+  stream_plan: |
+    StreamMaterialize { columns: [id1, a1, id2, a2, stream._row_id(hidden)], pk_columns: [stream._row_id, id2, id1], pk_conflict: "no check" }
+    └─StreamTemporalJoin { type: LeftOuter, predicate: stream.id1 = version.id2, output: [stream.id1, stream.a1, version.id2, version.a2, stream._row_id] }
+      ├─StreamExchange { dist: HashShard(stream.id1) }
+      | └─StreamTableScan { table: stream, columns: [stream.id1, stream.a1, stream._row_id], pk: [stream._row_id], dist: UpstreamHashShard(stream._row_id) }
+      └─StreamNoShuffleExchange { dist: UpstreamHashShard(version.id2) }
+        └─StreamTableScan { table: version, columns: [version.id2, version.a2], pk: [version.id2], dist: UpstreamHashShard(version.id2) }
```

Comment:
IIUC, we need this child only for...
- sync the look-up with an inner-side barrier,
- update the in-memory cache.

Reply:
Another important reason is that we need to enforce the scheduler to place `StreamTemporalJoin` with the inner side `StreamTableScan ` so that it can look up the inner side table at the CN.

Reply:
Makes sense. 🥰

Reply:
A `StreamNoShuffleExchange` would be converted into a `NoShuffle` Edge ultimately, linking the temporal join fragment and the scan fragment, and then the scheduler will figure out the scheduling thing for us.